model/lifecycle.go

```diff
@@ -1355,6 +1345,42 @@ func getContainerFromRunOn(id string, buildVarTask BuildVariantTaskUnit, buildVa
 	return container, nil
 }
 
+// getContainerOptions resolves the task's container configuration based on the
+// task's container name and the container definitions available to the project.
+func getContainerOptions(project *Project, pRef *ProjectRef, container string) (*task.ContainerOptions, error) {
+	for _, c := range project.Containers {
+		if c.Name != container {
+			continue
+		}
+
+		opts := task.ContainerOptions{
+			WorkingDir:     c.WorkingDir,
+			Image:          c.Image,
+			OS:             c.System.OperatingSystem,
+			Arch:           c.System.CPUArchitecture,
+			WindowsVersion: c.System.WindowsVersion,
+		}
+
+		if c.Resources != nil {
+			opts.CPU = c.Resources.CPU
+			opts.MemoryMB = c.Resources.MemoryMB
+			return &opts, nil
+		}
+
+		for name, size := range pRef.ContainerSizes {
```

Comment:
ContainerSizes is a map so I think we can just do a typical `containerSize, ok := pRef.ContainerSizes[c.Size]` instead

Reply:
Ah yeah, good catch. Fixed.